server/lib/flow-metrics.js

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var crypto = require('crypto');
+
+module.exports = function flowEventData(config, request) {
+
+  var flowId = crypto.randomBytes(16).toString('hex');
+  var flowBeginTime = Date.now();
+
+  // Incorporate request metadata into the flow id, so that
+  // receiving servers can cross-check the metrics bundle.
+  var flowSignature = flowId + '\n';
+  flowSignature += new Buffer('' + flowBeginTime).toString('hex') + '\n';
+  flowSignature += request.headers['user-agent'];
+
+  var key = config.get('flow_id_key');
+  flowSignature = crypto.createHmac('sha256', key).update(flowSignature);
```

Comment:
Correct; or at least, it will be, and we'll log metrics about it in order to cross-check the flow of metrics from content to auth server.